iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ValidationUtil.java

```diff
@@ -424,4 +586,149 @@ public static void checkLossFactor(Validable validable, float lossFactor) {
+        validationLevel = ValidationLevel.min(validationLevel, checkTargetDeadband(validable, "phase tap changer", ptc.isRegulating(), ptc.getTargetDeadband(), throwException, reporter));
+        return validationLevel;
+    }
+
+    private static ValidationLevel checkThreeWindingsTransformer(Validable validable, ThreeWindingsTransformer twt, boolean throwException, Reporter reporter) {
+        ValidationLevel validationLevel = ValidationLevel.LOADFLOW;
+        for (ThreeWindingsTransformer.Leg leg : twt.getLegs()) {
+            if (leg.hasRatioTapChanger()) {
+                validationLevel = ValidationLevel.min(validationLevel, checkRtc(validable, leg.getRatioTapChanger(), twt.getNetwork(), throwException, reporter));
+            }
+            if (leg.hasPhaseTapChanger()) {
+                validationLevel = ValidationLevel.min(validationLevel, checkPtc(validable, leg.getPhaseTapChanger(), twt.getNetwork(), throwException, reporter));
+            }
+        }
+        long regulatingTc = twt.getLegStream()
+                .map(ThreeWindingsTransformer.Leg::getRatioTapChanger)
+                .filter(Objects::nonNull)
+                .filter(TapChanger::isRegulating)
+                .count()
+                + twt.getLegStream()
+                .map(ThreeWindingsTransformer.Leg::getPhaseTapChanger)
+                .filter(Objects::nonNull)
+                .filter(TapChanger::isRegulating)
+                .count();
+        if (regulatingTc > 1) {
+            throw new ValidationException(validable, UNIQUE_REGULATING_TAP_CHANGER_MSG);
+        }
+        return validationLevel;
+    }
+
+    private static ValidationLevel checkTwoWindingsTransformer(Validable validable, TwoWindingsTransformer twt, boolean throwException, Reporter reporter) {
+        ValidationLevel validationLevel = ValidationLevel.LOADFLOW;
+        if (twt.hasRatioTapChanger()) {
+            validationLevel = ValidationLevel.min(validationLevel, checkRtc(validable, twt.getRatioTapChanger(), twt.getNetwork(), throwException, reporter));
+        }
+        if (twt.hasPhaseTapChanger()) {
+            validationLevel = ValidationLevel.min(validationLevel, checkPtc(validable, twt.getPhaseTapChanger(), twt.getNetwork(), throwException, reporter));
+        }
+        if (twt.getOptionalRatioTapChanger().map(RatioTapChanger::isRegulating).orElse(false)
+                && twt.getOptionalPhaseTapChanger().map(PhaseTapChanger::isRegulating).orElse(false)) {
+            throw new ValidationException(validable, UNIQUE_REGULATING_TAP_CHANGER_MSG);
+        }
+        return validationLevel;
+    }
+
+    private static ValidationLevel checkIdentifiable(Identifiable<?> identifiable, ValidationLevel previous, boolean throwException, Reporter reporter) {
```

Comment:
Code is really hard to read with all these min(). I have nothing to propose right now but I wish we could find a simpler way to deal with this.

Reply:
Since there are only two validation levels at the moment, I think we can simply return SCADA when a check is not LOADFLOW